cmd/reflect/reflect.go

```diff
@@ -0,0 +1,160 @@
+// Copyright 2020 Authors of Hubble
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package reflect
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/cilium/hubble/pkg/defaults"
+	"github.com/golang/protobuf/proto"
+	"github.com/golang/protobuf/protoc-gen-go/descriptor"
+	"github.com/spf13/cobra"
+	"google.golang.org/grpc"
+	rpb "google.golang.org/grpc/reflection/grpc_reflection_v1alpha"
+)
+
+var (
+	serverURL string
+)
+
+// New returns the reflect command.
+func New() *cobra.Command {
+	reflectCmd := &cobra.Command{
+		Use:   "reflect",
+		Short: "Let Hubble reflect upon itself",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runReflect()
+		},
+		Hidden: true,
+	}
+	reflectCmd.Flags().StringVarP(&serverURL, "server", "", defaults.GetDefaultSocketPath(), "URL to connect to server")
+	return reflectCmd
+}
+
+func runReflect() (err error) {
+	conn, err := grpc.Dial(serverURL, grpc.WithInsecure())
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if closeErr := conn.Close(); closeErr != nil && err == nil {
+			err = closeErr
+		}
+	}()
```

Comment:
Since #369, we don't duplicate code to create a new `conn` and use `cmd/common`.  PR #372 changes this a bit as well to ensure consistent settings are passed when creating a new `conn`. 

The TLDR is that you should init the conn in `RunE` and pass it down to `runReflect`. It would look something like this:
```go
import "github.com/cilium/hubble/cmd/common/conn"

func New(vp *viper.Viper) *cobra.Command {
	return := &cobra.Command{
		Use:   "reflect",
		Short: "Let Hubble reflect upon itself",
		RunE: func(_ *cobra.Command, _ []string) error {
                        ctx, cancel := context.WithCancel(context.Background())
                        defer cancel()
                        hubbleConn, err := conn.New(ctx, vp.GetString("server"), vp.GetDuration("timeout"))
                        if err != nil {
                                return err
                        }
                        defer hubbleConn.Close()
                        return runReflect(ctx, hubbleConn)
		},
		Hidden: true,
	}

```